app/helpers/error_helper.rb

```diff
@@ -29,6 +29,10 @@ def self.invalid_project_id(sample)
     def self.missing_required_metadata(sample, missing_metadata_fields)
       "Could not save sample '#{sample['name']}'. Missing required metadata: #{missing_metadata_fields.join(', ')}"
     end
+
+    def self.missing_input_files_or_token(sample_name)
+      "Neither basespace_access_token nor input_files_attributes found in sample #{sample_name}"
```

Comment:
why not split this into two different error messages?

Reply:
Because you need one OR the other. I've changed the wording of the message to be a little clearer.